power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -113,11 +113,14 @@ inline void process_trafo3w_edge(main_core::main_model_state_c auto const& state
             auto const& non_tap_side_node = tap_at_first_side ? to_node : from_node;
             // add regulated idx only when the first side node is tap side node.
             // This is done to add only one directional edge with regulated idx.
-            Idx2D const regulated_idx = from_node == tap_side_node ? unregulated_idx : trafo3w_idx;
-            add_to_edge(state, edges, edge_props, tap_side_node, non_tap_side_node, {regulated_idx, 1});
+            if (from_node == tap_side_node) {
+                add_to_edge(state, edges, edge_props, tap_side_node, non_tap_side_node, {unregulated_idx, 0});
+            } else {
+                add_to_edge(state, edges, edge_props, tap_side_node, non_tap_side_node, {trafo3w_idx, 1});
```

Comment:
Why this change to if-else? Is it because unlike the `unregulated_idx`, the `trafo3w_idx = 1` for the ranking weight (didn't change)?

Reply:
Addressed. For readability, a variable is used.